### ISSUE TITLE
Implement CET in kernel

### ISSUE THREAD

**Issue by kees** [KSPP/linux#142]
The x86 CET support is still being developed for userspace, but the kernel will need this defense internally as well.

**Comment by dragonsblade** [comment]
Does `-fcf-protection` in GCC help with this?

**Comment by kees** [comment]
> Does `-fcf-protection` in GCC help with this?

Yes, but the kernel itself needs to be prepared for dealing with the added instrumentation, etc.

**Comment by kees** [comment]
Start by reverting commit 20bf2b378729c4a0366a53e2018a0b70ace94bcd and properly fix tracing issues...
